lib/beaker-pe/install/pe_utils.rb

```diff
@@ -891,6 +891,245 @@ def config_hosts_for_proxy_access hosts
           end
         end
 
+        # removes the install_puppet_agent_pe_promoted_repo_on method
+
+        # Install shared repo of the puppet-agent on the given host(s).  Downloaded from
+        # location of the form PE_PROMOTED_BUILDS_URL/PE_VER/puppet-agent/AGENT_VERSION/repo
+        #
+        # @param [Host, Array<Host>, String, Symbol] hosts    One or more hosts to act upon,
+        #                            or a role (String or Symbol) that identifies one or more hosts.
+        # @param [Hash{Symbol=>String}] opts An options hash
+        # @option opts [String] :puppet_agent_version The version of puppet-agent to install, defaults to 'latest'
+        # @option opts [String] :pe_ver The version of PE (will also use host['pe_ver']), defaults to '4.0'
+        # @option opts [String] :copy_base_local Directory where puppet-agent artifact
+        #                       will be stored locally
+        #                       (default: 'tmp/repo_configs')
+        # @option opts [String] :copy_dir_external Directory where puppet-agent
+        #                       artifact will be pushed to on the external machine
+        #                       (default: '/root')
+        # @option opts [String] :puppet_collection Defaults to 'PC1'
+        # @option opts [String] :pe_promoted_builds_url Base URL to pull artifacts from
+        #
+        # @note on windows, the +:ruby_arch+ host parameter can determine in addition
+        #   to other settings whether the 32 or 64bit install is used
+        #
+        # @example
+        #   install_puppet_agent_pe_promoted_repo_on(host, { :puppet_agent_version => '1.1.0.227', :pe_ver => '4.0.0-rc1'})
+        #
+        # @return nil
+
+        # Gets host-specific information for PE promoted puppet-agent packages
+        #
+        # @param [Host] host The node to install agent on
+        # @param [String] puppet_collection Name of the puppet collection to use
+        # @param [Hash{Symbol=>String}] opts Options hash to provide extra values
+        #
+        # @return [String, String, String] Host-specific information for packages
+        #   1. release_path_end Suffix for the release_path. Used on Windows. Check
+        #   {Windows::Pkg#pe_puppet_agent_promoted_package_info} to see usage.
+        #   2. release_file Path to the file on release build servers
+        #   3. download_file Filename for the package itself
+        def pe_puppet_agent_promoted_package_info(host, puppet_collection = nil, _opts = {} )
```

Comment:
Since we are no longer relying on host type dispatch for the flavor of this method, we likely need to pull in the variations in Beaker::Host::{Mac/Windows}::Pkg and dispatch internally here.

Reply:
I've got a dispatching method, testing now. 

```ruby
        def pe_puppet_agent_promoted_package_info_dispatch(host, puppet_collection = nil, opts = {})
          case host.class
          when Beaker::Windows::Host
            return pe_puppet_agent_promoted_package_info_windows(host, puppet_collection, opts)
          when Beaker::Mac::Host
            return pe_puppet_agent_promoted_package_info_mac(host, puppet_collection, opts)
          when Beaker::Unix::Host
            return pe_puppet_agent_promoted_package_info_unix(host, puppet_collection, opts)
          end
          raise RuntimeError("#{host.class} not one of Beaker::(Windows|Mac|Unix)::Host")
        end
```

Reply:
This is wrong, the `Beaker::` prefix is redundant. New PR incoming soon